lib/aescrypt.rb

```diff
@@ -74,10 +84,23 @@ def self.decrypt_data(encrypted_data, key, iv, cipher_type)
   #:arg: iv => String
   #:arg: cipher_type => String  
   def self.encrypt_data(data, key, iv, cipher_type)
-    aes = OpenSSL::Cipher::Cipher.new(cipher_type)
+    self.warn_if_ecb cipher_type
+
+    aes = OpenSSL::Cipher.new(cipher_type)
     aes.encrypt
     aes.key = key
```

Comment:
Perhaps run the key derivation function (mentioned above) here, so that all uses of `encrypt_data` are safe, not just the simple uses. But perhaps you want the advanced functions to stay advanced. So alternatively assert that they key is non-nil and of the cipher's block length (some ciphers require different lengths).